pypsa/io.py

```diff
@@ -581,7 +579,11 @@ def _import_from_importer(network, importer, basename, skip_time=False):
     df = importer.get_snapshots()
     if df is not None:
         network.set_snapshots(df.index)
-        if "weightings" in df.columns:
+        cols = ['objective', 'generators', 'stores']
+        if not df.columns.intersection(cols).empty:
+            network.snapshot_weightings = df.reindex(
+                index=network.snapshots, columns=cols)
```

Comment:
This catches a case where not all of the three weightings are set, right? And fills the missing columns with `pd.NA`. Ok.

Reply:
Yes, I did not change the functionality at this point.